File: clients/app/android/app/src/main/java/com/petsitter/data/remote/dto/UserDto.kt
package com.petsitter.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// UserDto: Zod userSchema 대응
// kotlinx.serialization으로 JSON 직렬화/역직렬화
@Serializable
data class UserDto(
    val id: String,
    val email: String,
    val name: String?,
    val role: String,
    @SerialName("profileImage") val profileImage: String? = null,
)

@Serializable
data class LoginResponseDto(
    @SerialName("access_token")  val accessToken: String,
    @SerialName("refresh_token") val refreshToken: String,
    val user: UserDto,
)

@Serializable
data class LoginRequestDto(
    val email: String,
    val password: String,
)

@Serializable
data class RegisterRequestDto(
    val email: String,
    val password: String,
    val name: String,
    val role: String,
)

File: clients/app/ios/PetSitter/Data/DTO/UserDTO.swift
import Foundation

// ─── 요청 DTO ───────────────────────────────────────────────────────────────

struct LoginRequestDTO: Encodable {
    let email: String
    let password: String
}

struct RegisterRequestDTO: Encodable {
    let email: String
    let password: String
    let name: String
    let role: String
}

// ─── 응답 DTO ───────────────────────────────────────────────────────────────

struct LoginResponseDTO: Decodable {
    let accessToken: String
    let refreshToken: String
}

struct RefreshResponseDTO: Decodable {
    let accessToken: String
    let newRefreshToken: String
}

struct UserDTO: Decodable {
    let id: String
    let email: String
    let name: String?
    let role: String
    let profileImage: String?

    // DTO → Domain 변환
    func toDomain() -> User {
        User(
            id: id,
            email: email,
            name: name ?? "",
            role: UserRole(rawValue: role) ?? .petOwner,
            profileImage: profileImage
        )
    }
}
